### PR TITLE
Update minio-go library, startup banner and response meta for storage class

### PR DESCRIPTION
**Pull request by nitisht**
## Description
This PR fixes following storage class related issues

- Update startup banner to print storage class in capitals. This
makes it easier to identify different storage classes available.

- Update response metadata to not send STANDARD storage class.
This is in accordance with AWS S3 behaviour.

- Update minio-go library to bring in storage class related
changes. This is needed to make transparent translation of
storage class headers for Minio S3 Gateway.

## Motivation and Context
This is based on discussion after storage-class support was added to Minio server.

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by krisis — DISMISSED** [review]

**Comment by kannappanr** [comment]
@nitisht Does this change require mint testing? If yes can you please update the mint test results?

**Comment by harshavardhana** [comment]
> @nitisht Does this change require mint testing? If yes can you please update the mint test results?

mint test is added as part of minio-go functional tests, so we need to make new release to post results.

File: cmd/object-api-utils_test.go [hunk 1]
```diff
@@ -197,3 +198,94 @@ func TestIsMinioMetaBucketName(t *testing.T) {
...
+			want:     map[string]string{"content-type": "application/octet-stream", "etag": "de75a98baf2c6aef435b57dd0fc33c86"},
+		},
+	}
+	for _, tt := range tests {
+		if got := removeStandardStorageClass(tt.metadata); !reflect.DeepEqual(got, tt.want) {
+			t.Errorf("Test %s failed, expected %v, got %v", tt.name, tt.want, got)
+		}
+	}
+}
+
```
Comment (harshavardhana) [inline cmd/object-api-utils_test.go:232]: Add comments for each tests @nitisht

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/object-api-utils_test.go (hunk 1 above)
Comment (nitisht) [reply cmd/object-api-utils_test.go:232]: added comments @harshavardhana

File: cmd/object-api-utils.go [hunk 2]
```diff
@@ -187,14 +187,26 @@ func getCompleteMultipartMD5(parts []CompletePart) (string, error) {
...
+// Filter X-Amz-Storage-Class field only if it is set to STANDARD.
+// This is done since AWS S3 doesn't return STANDARD Storage class as response header.
+func removeStandardStorageClass(metadata map[string]string) map[string]string {
+	if metadata[amzStorageClass] == standardStorageClass {
+		delete(metadata, amzStorageClass)
+	}
+	return metadata
 }
 
 // Clean metadata takes keys to be filtered
```
Comment (krisis) [inline cmd/object-api-utils.go]: comment needs to be updated. s/Clean metadata/cleanMetadataKeys .../

**Review by krisis — COMMENTED** [review]
Changes LGTM, one minor comment.

File: cmd/object-api-utils.go (hunk 2 above)
Comment (nitisht) [reply cmd/object-api-utils.go]: updated the comment @krisis

**Review by harshavardhana — APPROVED** [review]
LGTM, tested as well locally.

**Review by krisis — APPROVED** [review]

**Comment by nitisht** [comment]
Codecov seems to be stuck, but unit tests are added in this PR, so taking this in.

**Pull request merged**